src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -818,20 +819,46 @@ public RequestLineStatus TakeStartLine(SocketInput input)
                     }
                 }
 
-                string method;
-                var begin = scan;
-                if (!begin.GetKnownMethod(out method))
+                end.Take();
+
+                var methodStart = scan;
+                var methodEnd = methodStart;
+                if (methodEnd.Seek(ref _vectorSpaces, ref end) == -1)
                 {
-                    if (scan.Seek(ref _vectorSpaces, ref end) == -1)
-                    {
-                        RejectRequest(RequestRejectionReason.MissingSpaceAfterMethod);
-                    }
+                    RejectRequest(RequestRejectionReason.InvalidRequestLine, GetInvalidRequestLineString(start, end));
```

Comment:
We might want to make sure that the logging level is high enough before calling GetInvalidRequestLineString.